static/src/javascripts/projects/common/modules/ui/subscription-banner.js

```diff
@@ -0,0 +1,163 @@
+// @flow
+
+import uniq from 'lodash/uniq';
+import { hasUserAcknowledgedBanner } from 'common/modules/ui/message';
+// import { trackNonClickInteraction } from 'common/modules/analytics/google';
+import config from 'lib/config';
+import userPrefs from 'common/modules/user-prefs';
+import type { Banner } from 'common/modules/ui/bannerPicker';
+import { local } from 'lib/storage';
+import {
+    submitViewEvent,
+    submitClickEvent,
+} from 'common/modules/commercial/acquisitions-ophan';
+import { shouldHideSupportMessaging } from 'common/modules/commercial/user-features';
+import { pageShouldHideReaderRevenue } from 'common/modules/commercial/contributions-utilities';
+import { track as trackFirstPvConsent } from 'common/modules/ui/first-pv-consent-banner';
+import {
+    setAdConsentState,
+    allAdConsents,
+} from 'common/modules/commercial/ad-prefs.lib';
+import { bannerTemplate } from 'common/modules/ui/subscription-banner-template';
+
+const messageCode = 'subscription-banner';
+const pageviews = local.get('gu.alreadyVisited');
+const subsciptionBannerClosedKey = 'subscriptionBannerLastClosedAt';
+const subscriptionBannerSwitch = config.get('switches.subscriptionBanner');
+const edition = config.get('page.edition');
+const subscriptionUrl =
+    'https://support.theguardian.com/subscribe/digital?INTCMP=gdnwb_copts_banner_subscribe_SubscriptionBanner&acquisitionData=%7B%22%3A%22GUARDIAN_WEB%22%2C%22campaignCode%22%3A%22subscriptions_banner%22%2C%22componentType%22%3A%22ACQUISITIONS_SUBSCRIPTIONS_BANNER%22%7D';
+const signInUrl =
```

Comment:
Might be nicer to make these URLs not tied to environment (e.g. would work on CODE/local too).  The environment dependent hosts can be grabbed from config, and then you'd need to append the path/query params.

```
const subscriptionHostname = config.get('page.supportUrl')
const signinHostname = config.get('page.idUrl')
```

Reply:
add these host names to make the urls work with all domains